pangeo_forge_recipes/recipes/xarray_zarr_beam.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import Any, Optional, Tuple
+
+import apache_beam as beam
+
+from ..patterns import Index
+from ..storage import CacheFSSpecTarget, OpenFileType, get_opener
+
+logger = logging.getLogger(__name__)
+
+
+IndexedElement = Tuple[Index, Any]
+# can we use a generic, e.g Indexed[xr.Dataset]?
```

Comment:
Yes, you can!  :) 

https://beam.apache.org/documentation/sdks/python-type-safety/#defining-generic-types